CRM/Core/BAO/CustomGroup.php

```diff
@@ -142,14 +142,16 @@ public static function create(&$params) {
       $group->created_id = $params['created_id'] ?? NULL;
       $group->created_date = $params['created_date'] ?? NULL;
 
-      // we do this only once, so name never changes
-      if (isset($params['name'])) {
-        $group->name = CRM_Utils_String::munge($params['name'], '_', 64);
+      // Process name only during create, so it never changes
+      if (!empty($params['name'])) {
+        $group->name = CRM_Utils_String::munge($params['name']);
       }
       else {
-        $group->name = CRM_Utils_String::munge($group->title, '_', 64);
+        $group->name = CRM_Utils_String::munge($group->title);
```

Comment:
Just noting the default max length is 63 so this will work with the current table definition

Reply:
Yes this PR changes it from 64 down to 63 (which happens to be the default for that function) to accommodate the potential extra character inserted during validation.